### ISSUE TITLE
Merge DM-stack setup into SN setup

### ISSUE THREAD

**Issue by RickKessler** [LSSTDESC/desc-help#78]
**Description**
A clear and concise description of what the issue is.  

Need to merge “/global/cscratch1/sd/heatherk/sprint-sn/setup_sn_some.sh” into the nominal SN-setup, or allow a flag, e.g., 
   source  /global/cfs/cdirs/lsst/groups/SN/software/setup_sn_all.sh dm-stack
More people need avro/dmStack, and this two-part setup is tripping people.

Choose all applicable topics by placing an 'X' between the [ ]:
- [ ] jupyter
- [ ] jupyter terminal
- [ ] Cori interactive command line
- [ ] Batch jobs
- [ ] python
- [ ] CSCRATCH
- [x ] Community File System
- [ ] HPSS (tape)
- [ ] Data transfer and Globus
- [ ] New User Account or account access problems

**To Reproduce**

**Screenshots**

**Comment by heather999** [comment]
@RickKessler Think this is now done. Also going to try keeping the setup scripts under some version management in github.

There is an updated [setup_sn_all.sh here](https://github.com/LSSTDESC/sn_env/blob/main/nersc/setup_sn_all.sh).
What would be the best way to test this out and inform folks before we swap the old version out and add this one?

The script now accepts an optional "lsst" parameter - so users would either do:
`source setup_sn_all.sh` as before and get the original setup 
or
`source setup_sn_all.sh lsst`  and get the LSST Science Pipelines environment along with the packages we set up for the Sprint in Nov.

A final comment:
* Right now, we've hardcoded w_2021_40 LSST Science Pipelines.  Maybe that's fine, not sure if you guys need to keep up with the more recent weekly builds or how that should be handled.

**Comment by heather999** [comment]
ah and while I'm thinking about it.. we may want to include a `perlmutter` option in this script, but that may require setting up SNANA on the Perlmutter machines.

**Comment by RickKessler** [comment]
in addition to the current /global/cfs/cdirs/lsst/groups/SN/software/setup_sn_all.sh
how about adding the new one as
/global/cfs/cdirs/lsst/groups/SN/software/setup_sn_all_test.sh

I will use the "test" version, and after running some tests I will move it to default for everyone else.

hard-coding "w_2021_40 LSST Science Pipelines" seems fine ... we can do something more elegant later ... if needed.

**Comment by heather999** [comment]
Ok - there's a new test script available:
` /global/cfs/cdirs/lsst/groups/SN/software/setup_sn_all_test.sh`

**Comment by RickKessler** [comment]
I am now using the test script in daily life.

**Comment by RickKessler** [comment]
with new setup script, GSL_DIR is not defined and thus make failed; 
GSL_DIR should be
/global/common/sw/cray/cnl7/haswell/gsl/2.5/gcc/8.2.0/sr445ay

**Comment by heather999** [comment]
Hi Rick,
Actually nothing has changed for the original set up script.. so when you run `source setup_sn_all.sh`, you have GSL_DIR defined, since we did a `module load gsl` to use NERSC's GSL libraries.
In the case of setting up the LSST Science Pipelines via `source setup_sn_all.sh lsst` - the LSST environment includes GSL 2.7 and we should not override that - the LSST code is using specific versions of these libraries - just like cfitsio.  It's true that GSL_DIR is left undefined though - I could set up NERSC's gsl library just as we did with cfitsio - but I'm a bit concerned since now we're dealing with a different version.  I don't think we want both versions of gsl in the same environment. Alternatively I could just define GSL_DIR and point at the gsl 2.7 that comes with the LSST conda-forge environment, but I'm not sure if you specifically need 2.5 for the other SN specific code.
